src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexInterpreter.cs

```diff
@@ -207,14 +207,14 @@ private void StackPush(int i1, int i2)
 
         private int Forwardchars() => _rightToLeft ? runtextpos - runtextbeg : runtextend - runtextpos;
 
-        private char Forwardcharnext()
+        private char Forwardcharnext(ReadOnlySpan<char> runtextSpan)
         {
-            char ch = _rightToLeft ? runtext![--runtextpos] : runtext![runtextpos++];
+            char ch = _rightToLeft ? runtextSpan![--runtextpos] : runtextSpan![runtextpos++];
```

Comment:
Separately, it might be worth doing a perf test where we measure changing this to:
```C#
int i = _rightToLeft ? --runtextpos : runtextpos++;
char ch = runtextspan[i];
return _caseInsensitive ? _textInfo.ToLower(ch) : ch;
```
I suspect that'll result in better codegen, and this is on a hot path for the interpreter.

Reply:
Ok, so I wrote a quick perf test which tries to match an email on a string like: 
```c#
Regex.Match(@"Try to capture an email here like somemail25@gmail.com for example.", @"\b[^@\s]+@[^@\s]+\.[^@\s]+\b");
```

The results without the optimization are:
```
|         Method |     Mean |     Error |    StdDev |
|--------------- |---------:|----------:|----------:|
| GetMatchResult | 1.981 us | 0.0395 us | 0.0692 us |
```

The results with the optimization are:
```
|         Method |     Mean |     Error |    StdDev |
|--------------- |---------:|----------:|----------:|
| GetMatchResult | 1.903 us | 0.0289 us | 0.0284 us |
```

So that is roughly a 4% gain. I'll add this change to the PR, thanks for suggesting.